apps/web/src/components/settings/ProfileSettings.tsx

```diff
@@ -5,50 +5,74 @@ import { Button } from "@/components/shadcn/ui/button";
 import { Label } from "@/components/shadcn/ui/label";
 import { Textarea } from "@/components/shadcn/ui/textarea";
 import {
-	modifyRegistrationData,
+	modifyProfileData,
 	updateProfileImage,
 } from "@/actions/user-profile-mod";
 import { useUser } from "@clerk/nextjs";
 import { useAction } from "next-safe-action/hook";
 import { toast } from "sonner";
 import { useState } from "react";
 import { encodeFileAsBase64 } from "@/lib/utils/shared/files";
+import { Tag, TagInput } from "@/components/shadcn/ui/tag/tag-input";
+import { Loader2 } from "lucide-react";
+import { Avatar, AvatarImage } from "@/components/shadcn/ui/avatar";
 
-interface ProfileSettingsProps {
+interface ProfileData {
+	pronouns: string;
 	bio: string;
-	university: string;
+	skills: string[];
+	discordUsername: string;
+	profilePhoto: string;
 }
 
-export default function ProfileSettings({
-	bio,
-	university,
-}: ProfileSettingsProps) {
-	const [newBio, setNewBio] = useState(bio);
-	const [newUniversity, setNewUniversity] = useState(university);
+interface ProfileSettingsProps {
+	profile: ProfileData;
+}
+
+export default function ProfileSettings({ profile }: ProfileSettingsProps) {
+	const [newPronouns, setNewPronouns] = useState<string>(profile.pronouns);
+	const [newBio, setNewBio] = useState(profile.bio);
 	const [newProfileImage, setNewProfileImage] = useState<File | null>(null);
+	let curSkills: Tag[] = [];
+	for (let i = 0; i < profile.skills.length; i++) {
```

Comment:
non-blocking: For future reference, it is preferred to use the built-in iterator functions as much as possible such as `map, forEach, Entries, etc`